src/tests/async/struct.cs

```diff
@@ -13,7 +13,13 @@ public class Async2Struct
     [Fact]
     public static void TestEntryPoint()
     {
-        Async().Wait();
+        // TODO: need to fix this
+
+        // Hits an assert around:
```

Comment:
This is another case that works with Await folding and trips over if optimization is disabled.

Like with the other TODO, this case can be hit regardless of optimization with directed scenario.
I will uncomment and log an issue.

Reply:
https://github.com/dotnet/runtimelab/issues/3011